### PR TITLE
Proposed other costructions, vocabulary and punctuation in a few places.

### PR DESCRIPTION
**Pull request by mradecki**
*   I was not sure about the meaning of callback related sentence in 220_Designing_for_composabiity_interfaces.md so I wrapped it in `?`'s.

*   Generally speaking, I might have misunderstood something, so double check all these ;).

### CODE REVIEW

File: manuscript/220_Designing_for_composabiity_interfaces.md [hunk 1]
```diff
@@ -74,7 +74,7 @@ public void SendSomethingToRecipients()
...
+This does not mean that events or callbacks are bad. It's just that they are not fit for replacing interfaces -- in reality, their purpose is a little bit different. We use events or callbacks not ?tell somebody to do something?, but to indicate what happened (that's why we call them events, after all...). This fits well the observer pattern we already talked about in the previous chapter. So, instead of using observer objects, we may consider using events or callbacks instead (as in everything, there are some tradeoffs for each of the solutions). In other words, events and callbacks have their use in the composition, but they are fit for a case so specific, that they cannot be treated as a default choice. The advantage of interfaces is that they bind together messages that represent a coherent abstractions and convey roles in the communication. This improves readability and clarity.
```
Comment (grzesiek-galezowski) [inline manuscript/220_Designing_for_composabiity_interfaces.md]: This is correct, I don't know how the earlier version creeped in :-). You can replace the question marks with quotes.

File: manuscript/230_Designing_for_composabiity_protocols.md [hunk 2]
```diff
@@ -487,7 +487,7 @@ public class TimedSession : Session
 
 and there is no need to change any other code to get this working[^statemachine]. 
 
-Another added bonus of this situation that `Session` does not have to return anything from methods is that we are free to apply proxy and decorator patterns more freely to the `Session`. For example, we may have hidden sessions, that are not displayed/stored/sent at all, but retain the rest of the session functionality. We may implement the feature using a proxy, that forwards all messages it receives to the original, wrapped `Session` object, but discards the `DumpInto()` calls: 
+Another added bonus of this situation is that `Session` does not have to return anything from methods is that we are free to apply proxy and decorator patterns more freely to the `Session`. For example, we may have hidden sessions, that are not displayed/stored/sent at all, but retain the rest of the session functionality. We may implement the feature using a proxy, that forwards all messages it receives to the original, wrapped `Session` object, but discards the `DumpInto()` calls:
```
Comment (grzesiek-galezowski) [inline manuscript/230_Designing_for_composabiity_protocols.md]: Actually, the "is" was already here, only after the "methods" word. I agree that the placement of "is" is a bit confusing, however, this doesn't fix the problem either. I suggest rewording it to:

"Another added bonus of the fact that `Session` methods don't have return types is that..."

What do you think?

File: manuscript/235_Classes.md [hunk 3]
```diff
@@ -18,7 +18,7 @@ A> A code of a Class should have only one reason to change.
 
 There has been a lot written about the principle on the web, so I am not going to be wiser than your favourite web search engine (my recent search yielded over 74 thousands results). Still, I believe it is useful to explain this principle in terms of composability.
 
-Usually, the hard part about this principle is how to understand "a reason to change". Robert C. Martin explains[^srponstackoverflow] that this is about a single source of entropy that generates changes to the class. Which leads us to another trouble of defining a "source of entropy". So I think it's better to just give you an example.
+Usually, the hard part about this principle is how to understand "a reason to change". Robert C. Martin explains[^srponstackoverflow] that this is about a single source of entropy that generates changes to the class, what leads us to another trouble of defining a "source of entropy". So I think it's better to just give you an example.
```
Comment (grzesiek-galezowski) [inline manuscript/235_Classes.md]: I don't think using "what" is more correct. What I wanted to use was something more like in the examples in http://pl.bab.la/slownik/angielski-polski/leads-me (there are some starting from "which")

File: manuscript/240_Refactoring_Object_Composition.md [hunk 4]
```diff
@@ -83,7 +83,7 @@ Let's sum up these two conclusions with the following statement:
...
 
-Wow, this is quite a statement, isn't it? But, as we already noticed, it is true. There is, however, one problem with treating the composition code as overall application description: readability. Even though the composition *is* the description of the system, it does not read naturally. We want to see the description of behavior, but most of what we see is: `new`, `new`, `new`, `new`, `new`... There is a lot of syntactic noise involved, especially in real systems, where composition code is much longer than this tiny example. Can't we do something about it?
+Wow, this is quite a statement, isn't it? But, as we already noticed, it is true. There is, however, one problem with treating the composition code as overall application description: readability. Even though the composition *is* the description of the system, it is not naturally readable. We want to see the description of behavior, but most of what we see is: `new`, `new`, `new`, `new`, `new`... There is a lot of syntactic noise involved, especially in real systems, where composition code is much longer than this tiny example. Can't we do something about it?
```
Comment (grzesiek-galezowski) [inline manuscript/240_Refactoring_Object_Composition.md]: "it does not read naturally" seems to be an acceptable form (see http://www.usingenglish.com/forum/threads/36826-The-book-reads-well) - is there a reason to change it?

**Comment by grzesiek-galezowski** [comment]
Many thanks for the pull request! There are some points I'd like to discuss - let's clear them up and I'll happily merge it!

File: manuscript/230_Designing_for_composabiity_protocols.md (hunk 2 above)
Comment (mradecki) [inline manuscript/230_Designing_for_composabiity_protocols.md]: Oh now I see that I misunderstood this sentence.

It is better now, however it is still weird somehow. Well, at least to me.

How about other approach, like: "Another advantage of designing/making `Session` to not return anything from its methods is"?

File: manuscript/235_Classes.md (hunk 3 above)
Comment (mradecki) [inline manuscript/235_Classes.md]: It seems you are right. I wasn't familiar with that construction.

File: manuscript/240_Refactoring_Object_Composition.md (hunk 4 above)
Comment (mradecki) [inline manuscript/240_Refactoring_Object_Composition.md]: Another new thing for me. But still, people on this forum discuss whether this phrase is colloquial or not. This makes me think that it is still not a good idea to use it into a technical book, but that's up to you.

File: manuscript/230_Designing_for_composabiity_protocols.md (hunk 2 above)
Comment (grzesiek-galezowski) [inline manuscript/230_Designing_for_composabiity_protocols.md]: Your version looks fine. Please change the text according to your proposal.

File: manuscript/240_Refactoring_Object_Composition.md (hunk 4 above)
Comment (grzesiek-galezowski) [inline manuscript/240_Refactoring_Object_Composition.md]: Let's leave the original version for now. I'll do some research on it and maybe change it later when reviewing style of this chapter (currently the review is at earlier chapters, but I'll eventually reach this one)

**Comment by grzesiek-galezowski** [comment]
Hi, there are only few changes pending before I can accept this pull request. If you make those changes, I will merge it with the mainline.

**Comment by mradecki** [comment]
Hi, Yeah I am on it. I was quite busy this week. I will fix these issues
during weekend.

2016-04-29 23:29 GMT+02:00 Grzegorz Gałęzowski <notifications@github.com>:

> Hi, there are only few changes pending before I can accept this pull
> request. If you make those changes, I will merge it with the mainline.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/grzesiek-galezowski/tdd-ebook/pull/88#issuecomment-215886685>
>



-- 
--

Pozdrawiam,

Marek Radecki

**Comment by grzesiek-galezowski** [comment]
@mradecki I'd like to release next version today and there are some nice changes in this pull request, so I'll merge your changes and make the corrections we agreed to myself. Hope you don't mind!

Thank you for your work!
